### PR TITLE
Node routing table, fix #47

### LINKED ISSUE #47: Rethink routing table in NODE mode to manage node ID.

### PR DESCRIPTION
**Pull request by nicolas-rabault**
Rethink node routing table to manage node_id and certificates

### CODE REVIEW

File: Robus/src/reception.c [hunk 1]
```diff
@@ -130,7 +130,7 @@ void Recep_GetData(volatile unsigned char *data)
                            ((unsigned short)current_msg->data[data_size + 1] << 8);
             if (crc == crc_val)
             {
-                if ((current_msg->header.target_mode == IDACK) && (current_msg->header.target != DEFAULTID))
+                if (((current_msg->header.target_mode == IDACK) | (current_msg->header.target_mode == NODEIDACK)) && (current_msg->header.target != DEFAULTID))
```
Comment (Salem-Luos) [inline Robus/src/reception.c:133]:
```suggestion
                if (((current_msg->header.target_mode == IDACK) || (current_msg->header.target_mode == NODEIDACK)) && (current_msg->header.target != DEFAULTID))
```

File: Robus/src/reception.c [hunk 2]
```diff
@@ -139,7 +139,7 @@ void Recep_GetData(volatile unsigned char *data)
             else
             {
                 ctx.status.rx_error = TRUE;
-                if ((current_msg->header.target_mode == IDACK))
+                if ((current_msg->header.target_mode == IDACK) | (current_msg->header.target_mode == NODEIDACK))
```
Comment (Salem-Luos) [inline Robus/src/reception.c:142]:
```suggestion
                if ((current_msg->header.target_mode == IDACK) || (current_msg->header.target_mode == NODEIDACK))
```

File: Robus/src/robus.c [hunk 3]
```diff
@@ -152,7 +150,7 @@ uint8_t Robus_SendMsg(vm_t *vm, msg_t *msg)
         }
     }
     // Check if ACK needed
-    if (msg->header.target_mode == IDACK)
+    if ((msg->header.target_mode == IDACK) | (msg->header.target_mode == NODEIDACK))
```
Comment (Salem-Luos) [inline Robus/src/robus.c:155]:
```suggestion
    if ((msg->header.target_mode == IDACK) || (msg->header.target_mode == NODEIDACK))
```

File: src/luos.c [hunk 4]
```diff
@@ -69,47 +69,60 @@ void Luos_Loop(void)
     {
         // There is a message available find the container linked to it
         container_t *container = Luos_GetContainer(oldest_vm);
-        // Is this container having a callback?
-        if (container->cont_cb != 0)
+        // check if this is a Luos Command
+        uint8_t cmd = 0;
+        uint16_t size = 0;
+        if (MsgAlloc_GetLuosTaskCmd(remaining_msg_number, &cmd) == FAIL)
```
Comment (Salem-Luos) [inline src/luos.c:75]:
```suggestion
        while (MsgAlloc_GetLuosTaskCmd(remaining_msg_number, &cmd) != FAIL);
```
just to have less ligne in luos loop not really important

File: src/luos.c [hunk 5]
```diff
@@ -69,47 +69,60 @@ void Luos_Loop(void)
...
         {
-            // This container have a callback pull the message
-            if (MsgAlloc_PullMsg(oldest_vm, &returned_msg) == SUCESS)
+            // this is a critical failure we should never go here
+            while (1)
+                ;
+        }
+        if (MsgAlloc_GetLuosTaskSize(remaining_msg_number, &size) == FAIL)
```
Comment (Salem-Luos) [inline src/luos.c:81]: same as last comment

File: Robus/src/robus.c [hunk 6]
```diff
@@ -81,7 +83,7 @@ void Robus_Loop(void)
  * @param type of container create
  * @return None
  ******************************************************************************/
-vm_t *Robus_ContainerCreate(unsigned char type)
+vm_t *Robus_ContainerCreate(unsigned short type)
```
Comment (Salem-Luos) [inline Robus/src/robus.c:86]:
```suggestion
vm_t *Robus_ContainerCreate(uint16_t type)
```

File: Robus/inc/context.h [hunk 7]
```diff
@@ -35,8 +35,7 @@ typedef struct
     // Variables
     DATA_CB data_cb;                  /*!< Data management callback. */
     status_t status;                  /*!< Status. */
-    unsigned short id;                /*!< Container ID. */
-    unsigned char type;               /*!< Container type. */
+    unsigned short node_id;           /*!< Node ID. */
```
Comment (Salem-Luos) [inline Robus/inc/context.h]:
```suggestion
    uint16_t node_id;           /*!< Node ID. */
```

File: src/luos.c (hunk 4 above)
Comment (nicolas-rabault) [reply src/luos.c]: I prefer to keep it as a if because it will be replaced by an assert one day...

**Review by Salem-Luos — APPROVED** [review]

**Pull request merged**